schema/5.1.0/schema.md

```diff
@@ -183,7 +183,8 @@ The following table describes the matrix data and layers requirements that are *
 | Assay | "raw" required? | "raw" location | "normalized" required? | "normalized" location |
 |-|-|-|-|-|
 | scRNA-seq (UMI, e.g. 10x v3, Slide-seqV2) | REQUIRED. Values MUST be de-duplicated molecule counts. Each cell MUST contain at least one non-zero value. All non-zero values MUST be positive integers stored as `numpy.float32`.| `AnnData.raw.X` unless no "normalized" is provided, then `AnnData.X` | STRONGLY RECOMMENDED | `AnnData.X` |
-| Visium Spatial Gene Expression | REQUIRED. It is STRONGLY RECOMMENDED to use the unfiltered feature-barcode matrix (<code>raw_feature_bc_matrix</code>). See <a href="https://www.10xgenomics.com/support/software/space-ranger/analysis/outputs/space-ranger-feature-barcode-matrices">Space Ranger Feature-Barcode Matrices</a>. Values MUST be de-duplicated molecule counts. All non-zero values MUST be positive integers stored as `numpy.float32`.| `AnnData.raw.X` unless no "normalized" is provided, then `AnnData.X` | STRONGLY RECOMMENDED | `AnnData.X` |
+| Visium Spatial Gene Expression | REQUIRED. Values MUST be de-duplicated molecule counts. All non-zero values MUST be positive integers stored as `numpy.float32`.<br><br>If <code>uns['spatial']['is_single']</code> is <code>False</code> then each cell MUST contain at least one non-zero value.<br><br>If <code>uns['spatial']['is_single']</code> is <code>True</code> then the unfiltered feature-barcode matrix (<code>raw_feature_bc_matrix</code>) MUST be used. See <a href="https://www.10xgenomics.com/support/software/space-ranger/analysis/outputs/space-ranger-feature-barcode-matrices">Space Ranger Feature-Barcode Matrices</a>. This matrix MUST contain 4992 rows. If the <code>obs['in_tissue']</code> value is <code>1</code>, then the cell MUST contain at least one non-zero value, If any <code>obs['in_tissue']</code> values are <code>0</code>, then at least one cell corresponding to a <code>obs['in_tissue']</code> with a value of <code>0</code> MUST contain a non-zero value.| `AnnData.raw.X` unless no "normalized" is provided, then `AnnData.X` | STRONGLY RECOMMENDED.<br>This MUST be a <a href="https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.html"><code>scipy.sparse.csr_matrix</code></a>. If the <code>obs['in_tissue']</code> is <code>0</code> for a cell, then values MUST be encoded as <a href="https://docs.scipy.org/doc/scipy/tutorial/sparse.html#sparse-arrays-implicit-zeros-and-duplicates">implicit zero</a>. | `AnnData.X` |
+|||||
```

Comment:
Why does this have additional specifications about sparse.csr_matrix & implicit zeros
(as opposed to other data which are just `In any layer, if a matrix has 50% or more values that are zeros, it is STRONGLY RECOMMENDED that the matrix be encoded as a scipy.sparse.csr_matrix`)?

Reply:
I was responding to Pablo's recommendation (which he [previously approved](https://github.com/chanzuckerberg/single-cell-curation/issues/674#issuecomment-2000514827) in its earlier iteration. 

Pablo Garcia-Nieto
(https://czi-sci.slack.com/archives/C02TSE92DBM/p1707268059182289?thread_ts=1707240003.577709&cid=C02TSE92DBM)
Ah great to hear! Census is not going to ingest normalized data, so no strong opinions on that. For all other uses I believe it’s ok to add these dummy 0s, but then I would strongly recommend to store these matrices in scipy sparse format and to encode the 0's implicitly. 

 **this should be enforced by the validator.**

Reply:
I think the original text pointed out by Jason with slight modification captures my recommendation:

> In any layer, if a matrix has 50% or more values that are zeros, it is STRONGLY RECOMMENDED that the matrix be encoded as a scipy.sparse.csr_matrix. **If the matrix is encoded as a scipy.sparse.csr_matrix, then zero values MUST be encoded as <a href="https://docs.scipy.org/doc/scipy/tutorial/sparse.html#sparse-arrays-implicit-zeros-and-duplicates">implicit zeros</a>.**

And then you can remove the following

> <br>This MUST be a <a href="https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.html"><code>scipy.sparse.csr_matrix</code></a>. If the <code>obs['in_tissue']</code> is <code>0</code> for a cell, then values MUST be encoded as <a href="https://docs.scipy.org/doc/scipy/tutorial/sparse.html#sparse-arrays-implicit-zeros-and-duplicates">implicit zero</a>.